modcc/printer/gpuprinter.cpp

```diff
@@ -384,11 +384,23 @@ void emit_api_body_cu(std::ostream& out, APIMethod* e, bool is_point_proc) {
     auto body = e->body();
     auto indexed_vars = indexed_locals(e->scope());
 
-    std::unordered_set<std::string> indices;
+    struct index_prop {
+        std::string source_var; // array holding the indices
+        std::string index_name; // index into the array
+
+        bool operator<(const index_prop& other) const {
```

Comment:
When this is inserted in a `std::set`, if you have two indexes with the same `source_var` and different `index_name`, then only one will be stored.
Will that situation ever arise? Or will a given `source_name` only ever have one possible `index_name`? If that's the case, then a hash table might be a better option.

Reply:
Reading the code a bit more, this approach has the benefit of printing all the variables in alphabetical (deterministic) order.

Reply:
Actually, I have a bug here that I hadn't noticed before. Alphabetical order is not enough, because we have scenarios such as:
```
auto node_index_i_ = node_index[i_];
auto vec_ci_i_ = vec_ci_[node_index_i_];
```
We need to make sure that `node_index_i` is printed before `vec_ci_i_` regardless of the names of the vectors. Basically, all arrays that are indexed directly with `i_` should go first. I have changed `std::set` to an `std::list` and I insert either to the front or the back of the list to enforce my ordering.